app/clients/zendesk.py

```diff
@@ -0,0 +1,96 @@
+import json
+from typing import Dict, List, Union
+from urllib.parse import urljoin
+
+import requests
+from flask import current_app
+from requests.auth import HTTPBasicAuth
+
+from app.user.contact_request import ContactRequest
+
+__all__ = ["Zendesk"]
+
+
+class Zendesk(object):
```

Comment:
Hmm why not reuse the [`ZenDeskSell` class](https://github.com/cds-snc/notification-api/blob/bebd806cad9581101e5cc1d52db2613c1baa1fa9/app/clients/zendesk_sell.py#L16) we already have defined? Is this a different API class than the Zendesk Sell API?

If that is the case, I wonder if we can have a base class for both types of API for Zendesk that would share common configuration and functions?

Reply:
Yes, I believe `Zendesk` varies from `ZenDeskSell`, I think we could have a parent class for both types of Zendesk APIs

Reply:
reviewed this and zendesk & zendesk sell have nothing in common. zendesk sell is an acquisition and is still using the [old companies api's](https://developers.getbase.com/docs/rest/articles/first_call). Considering we're only using this class for one call, abstracting and trying to identify a common element isn't practical. Recommending we leave as-is.

Reply:
My thinking for reusability was not so much on the business domain level but more on the technical one. There are some shared properties between the two classes such as the base URL and the api key. The `zendesk_sell` module also has a `_send_request` function that has better practices for error handling IMHO. For example, the original `freshdesk` module catch the exception in place, log it and re-throw, which would cause a logging of the same error twice.

Reply:
The main reason i recommended not creating a common class and mirroring `zendesk_sell` is that we only have one POST request at this time. I think it comes down to a preference on when we want to optimize and i believe it's premature for zendesk since the only shared functionality is the base url & api key which all API's have, in addition to having different authentication mechanisms.

If we add additional functionality then i believe refactoring `Zendesk` to use a single `_send_request` makes sense since you don't have to worry about how to authenticate, etc. I've also changed how the error handling is done and what i have is very similar to `zendesk_sell` with the main difference being that the request is done in a single function + throwing a custom exception

Reply:
The main reuse I would have liked is around a common `_send_request` where the error handling could have been the same, so there was more than base url and api key for reusability. I am okay with some duplicate logic though and given we'll decommission Freshdesk soon.